src/AuthenticationResponse.js

```diff
@@ -321,14 +356,25 @@ class AuthenticationResponse {
   /**
    * decodeIDToken
    *
-   * @param {Object} response
-   * @returns {Promise}
+   * Note: If the `id_token` is not present in params, this method does not
+   * get called (short-circuited in `validateIDToken()`).
+   *
+   * @param response {AuthenticationResponse}
+   * @param response.params {object}
+   * @param [response.params.id_token] {string} IDToken encoded as a JWT
+   *
+   * @returns {AuthenticationResponse} Chainable
    */
   static decodeIDToken (response) {
     let jwt = response.params.id_token
 
-    if (jwt) {
+    try {
```

Comment:
Do we want to get rid of the check? Missing jwt seems to warrant another error.

Reply:
I think we do. If `id_token` is missing, the flow gets short-circuited earlier in `validateIDToken()`. So this method will not be called.

Even for standalone reasons, if it's missing, that'll be caught in the `try` / `catch`